desk/app/activity.hoon

```diff
@@ -619,49 +619,57 @@
       %item
     =/  new-read  [id.action ~]
     =/  read-items  (put:on-read-items:a items.reads.index new-read)
-    =.  cor  (update-parents source ~[new-read])
-    (update-index source index(items.reads read-items) &)
+    =.  cor  (propagate-read-items source ~[new-read])
+    (refresh-index source index(items.reads read-items) &)
   ::
       %all
-    =/  new-reads=(list [=time-id:a ~])
-      %+  murn
-        %-  tap:on-event:a
-        (lot:on-event:a stream.index `floor.reads.index ~)
-      |=  [=time =event:a]
-      ?:  child.event  ~
-      `[time ~]
-    =/  new  index(items.reads (malt new-reads))
-    =?  cor  !from-parent
-      (update-parents source new-reads)
-    (update-index source new &)
-  ::
-      %recursive
-    =/  new-floor=time
+    =/  new=index:a
+      ::  if we're only marking at our level, then we only want to grab
+      ::  read items from our level, because we can't move the floor in
```

Comment:
```hoon
::  take every event between the floor and now, and put it into
::  the index's items.reads. this way, the floor can be moved without
::  "losing" any unreads, and the call to +refresh-index below will clean
::  up unnecessary items.reads entries.
::
```

Here or slightly further down.

Reply:
Also we should move the `?:  deep.action` branching to only branch once at the top of this function. More clear/coherent wrt guarantees about the changes we're making to the index.